common/djangoapps/student/models.py

```diff
@@ -1344,6 +1344,38 @@ def is_paid_course(self):
 
         return False
 
+    def is_active_course(self, cert_status):
+        """
+        `cert_status` is the certificate status for this enrollment
+        Returns (True) if end date of this course is in the future
+
+        """
+        if self.course_overview.end is None:
+            return True
+
+        time_delta = datetime.now(UTC)
+        if self.course_overview.end is not None and self.course_overview.end > time_delta:
+            return True
+
+        if self.course_overview.end is not None and self.course_overview.end < time_delta and cert_status['status'] == 'processing':
+            return True
+
+        return False
+
+    def is_completed_course(self, cert_status):
+        """
+        `cert_status` is the certificate status for this enrollment
+        Returns (True) if the course is successfully completed
+        """
+        time_delta = datetime.now(UTC)
+        if self.course_overview.end is not None and self.course_overview.end < time_delta and self.course_overview.may_certify() and cert_status['status'] != 'processing':
+            return True
+
+        if self.course_overview.may_certify() and cert_status['status'] in ('generating', 'ready', 'notpassing', 'restricted'):
```

Comment:
`in ('generating', 'ready', 'notpassing', 'restricted')` move this to some class constants don't hard code in this code.